From 56b42025528098d465ebe239f9c4a347c79a253f Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Wed, 30 Dec 2015 15:46:42 +0800
Subject: [PATCH 2/5] css: integrate isp2400 and 2401 ia_css_20151229_1822

Integrate FW binary isp2400 and 2401_legacy
for ia_css_20151229_1822 firmware

Change-Id: Iac48b6a10c293aed1c72d303701e3e6534881508
Original-Tracked-On: #H1504131695
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/92629
Integrator: WeiX Meng <weix.meng@intel.com>
Approver: Jing Tao <jing.tao@intel.com>
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-15122
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/455604
---
 .../camera/drivers/media/pci/atomisp2/css/ia_css.h             |  2 +-
 .../camera/drivers/media/pci/atomisp2/css/ia_css_types.h       |  2 +-
 .../drivers/media/pci/atomisp2/css/ia_css_version_data.h       |  2 +-
 .../drivers/media/pci/atomisp2/css/runtime/binary/src/binary.c | 10 ++++++++--
 .../camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c    |  4 ++--
 5 files changed, 13 insertions(+), 7 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
index 2d31e46..8594fa4f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20151227_0200 */
+/* Release Version: irci_ecr-master_20151229_1822 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
index ac2d03c..5a0b70c 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20151227_0200 */
+/* Release Version: irci_ecr-master_20151229_1822 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
index d033702..47d6bb2 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
@@ -22,7 +22,7 @@ more details.
 #define __IA_CSS_VERSION_DATA_H
 
 
-#define CSS_VERSION_STRING "REL:20151227_52.0_0939; API:2.1.20.20; GIT:irci___#8f56dd693e4bd755b3c56e1383dc2c9ced11462d; SDK:; USER:viedifw; "
+#define CSS_VERSION_STRING "REL:20151229_53.2_2234; API:2.1.20.20; GIT:irci____3#c6a33634a2caf2b12d3807c2438571e1790b0cd3; SDK:; USER:viedifw; "
 
 
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/src/binary.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/src/binary.c
index 147fbd6..47fca95 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/src/binary.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/src/binary.c
@@ -1324,8 +1324,14 @@ ia_css_binary_find(struct ia_css_binary_descr *descr,
 	if (mode == IA_CSS_BINARY_MODE_VIDEO) {
 		dvs_env = descr->dvs_env;
 		need_dz = descr->enable_dz;
-		/* Video is the only mode that has a nodz variant. */
-		need_dvs = dvs_env.width || dvs_env.height;
+		/* need_dvs is set to false explicitly for IR case, as IR does not support DVS
+		(enable_luma_only would be true only for IR case). Pipe config tool could not generate
+		non zero dvs_env.width and dvs_env.height,even when dvs was set to off. Refer #H1504113120*/
+		if (enable_luma_only)
+			need_dvs = false;
+		else
+			/* Video is the only mode that has a nodz variant. */
+			need_dvs = dvs_env.width || dvs_env.height;
 	}
 
 	/* print a map of the binary file */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
index 279413c..c616dde 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
@@ -47,10 +47,10 @@ struct fw_param {
 /* Warning: same order as SH_CSS_BINARY_ID_* */
 static struct firmware_header *firmware_header;
 
-/* The string STR(irci_ecr-master_20151227_0200) is a place holder
+/* The string STR(irci_ecr-master_20151229_1822) is a place holder
  * which will be replaced with the actual RELEASE_VERSION
  * during package generation. Please do not modify  */
-static const char *release_version = STR(irci_ecr-master_20151227_0200);
+static const char *release_version = STR(irci_ecr-master_20151229_1822);
 
 #define MAX_FW_REL_VER_NAME	300
 static char FW_rel_ver_name[MAX_FW_REL_VER_NAME] = "---";
-- 
1.9.1

